addons/delivery/models/delivery_grid.py

```diff
@@ -93,6 +93,8 @@ def _get_price_from_picking(self, total, weight, volume, quantity):
         price = 0.0
         criteria_found = False
         price_dict = {'price': total, 'volume': volume, 'weight': weight, 'wv': volume * weight, 'quantity': quantity}
+        if self.free_over and total >= self.amount:
```

Comment:
Normally it's better using float_compare when you compare two floats but here you cannot access the currency of the order to get the decimal precision and total is already converted in the decimal precision of the currency.
So it seems ok to me